sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py

```diff
@@ -149,35 +119,50 @@ class TextTranslationClient(ServiceClientGenerated):
     Combinations of endpoint and credential values:
     str + AzureKeyCredential - used custom domain translator endpoint
     str + TokenCredential - used for regional endpoint with token authentication
-    str + TranslatorCredential - used for National Clouds
+    str + TokenCredential + Region - used for National Clouds
     None + AzureKeyCredential - used for global translator endpoint with global Translator resource
-    None + Token - general translator endpoint with token authentication
-    None + TranslatorCredential - general translator endpoint with regional Translator resource
-
-    :param endpoint: Supported Text Translation endpoints (protocol and hostname, for example:
-         https://api.cognitive.microsofttranslator.com). Required.
-    :type endpoint: str
-    :param credential: Credential used to authenticate with the Translator service
-    :type credential: Union[AzureKeyCredential , TokenCredential , TranslatorCredential, TranslatorAADCredential]
-    :keyword api_version: Default value is "3.0". Note that overriding this default value may
+    None + TokenCredential - general translator endpoint with token authentication
+    None + TokenCredential + Region - general translator endpoint with regional Translator resource
+
+    :keyword str endpoint: Supported Text Translation endpoints (protocol and hostname, for example:
+         https://api.cognitive.microsofttranslator.com). If not provided, global translator endpoint will be used.
+    :keyword credential: Credential used to authenticate with the Translator service
+    :paramtype credential: Union[AzureKeyCredential, TokenCredential]
+    :keyword str region: Used for National Clouds.
+    :keyword str resource_id: Used with both a TokenCredential combined with a region.
+    :keyword  str api_version: Default value is "3.0". Note that overriding this default value may
      result in unsupported behavior.
-    :paramtype api_version: str
     """
 
+    @overload
     def __init__(
         self,
         *,
-        credential: Optional[Union[AzureKeyCredential, TokenCredential, TranslatorCredential, TranslatorAADCredential]] = None,
+        credential: Optional[AzureKeyCredential] = None,
```

Comment:
Should the credential be a required kw only in the overloads? Based on docstring, looks like you always need to pass one of TokenCredential or AzureKeyCredential